### ISSUE TITLE
docs: SwitchDemo accessibility text

### ISSUE THREAD

**Issue by brian-cirieco** [radix-ui/primitives#3102]
## Documentation

The [documentation](https://www.radix-ui.com/primitives/docs/components/switch) on the Switch component shows a demo that does not comply with the accessibility guidelines detailed on the [switch role requirements](https://www.w3.org/WAI/ARIA/apg/patterns/switch). I assumed that the `htmlFor` would associate the label's accessible text, but according to the guidelines:

> The switch has an accessible label provided by one of the following:
> - Visible text content contained within the element with role switch.
> - A visible label referenced by the value of [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set on the element with role switch.
> - [aria-label](https://w3c.github.io/aria/#aria-label) set on the element with role switch.

Following the demo's example, my team used `SwitchDemo` as an example for our Storybook. The story flagged an accessibility violation involving the Switch not having accessible text.

![image](https://github.com/user-attachments/assets/a75e95a5-a802-4e69-b030-2be624f2c72d)

### Relevant Radix Component(s)

`SwitchDemo` example on the docs. I'd suggest we add the aria-labelledby to the switch, and the corresponding ID to the label:

```tsx
const SwitchDemo = () => (
  <form>
    <div className="flex items-center">
      <label id="airplane-mode-label" className="text-white text-[15px] leading-none pr-[15px]" htmlFor="airplane-mode">
        Airplane mode
      </label>
      <Switch.Root
        aria-labelledby="airplane-mode-label"
        className="w-[42px] h-[25px] bg-blackA6 rounded-full relative shadow-[0_2px_10px] shadow-blackA4 focus:shadow-[0_0_0_2px] focus:shadow-black data-[state=checked]:bg-black outline-none cursor-default"
        id="airplane-mode"
        style={{ '-webkit-tap-highlight-color': 'rgba(0, 0, 0, 0)' }}
      >
        <Switch.Thumb className="block w-[21px] h-[21px] bg-white rounded-full shadow-[0_2px_2px] shadow-blackA4 transition-transform duration-100 translate-x-0.5 will-change-transform data-[state=checked]:translate-x-[19px]" />
      </Switch.Root>
    </div>
  </form>
);
```

**Comment by lucas-neuhaus-bairesdev** [comment]
Adding the `aria-labelledby` still leaves the component the accessibility issue of an input with no label (if the switch is inside of a form).

In my case I had to wrap the switch in the `<label />` to make the association implicit, since we don't have access to the `<input />`'s `id` field.
